### ISSUE TITLE
cmake FetchContent_Declare set TBB_TEST off failed

### ISSUE THREAD

**Issue by HEUBITLYJ** [oneapi-src/oneTBB#976]
i am trying to use tbb as a thirdparty，i use FetchContent_Declare to use tbb，here is my tbb.camke:
```
include(FetchContent)

FetchContent_Declare(
  tbb
  GIT_REPOSITORY https://github.com/oneapi-src/oneTBB.git
  GIT_TAG v2021.7.0
  GIT_SHALLOW ON
)

set(TBB_TEST OFF)

FetchContent_MakeAvailable(tbb)

```

but after cmake build, in cmakecache.txt,the TBB_TEST  option is still ON, is there something wrong with my tbb.cmake?

**Comment by isaevil** [comment]
@HEUBITLYJ `TBB_TEST` is declared via `option()` so all you need it to just change `set(TBB_TEST OFF)` to `option(TBB_TEST OFF)` in your CMake code.

**Comment by VincentRouvreau** [comment]
@HEUBITLYJ  and @isaevil I tried your proposal `option(TBB_TEST OFF)` in [this CMakeLists.txt](https://github.com/VincentRouvreau/onetbb_cmake_fetch_content/blob/62576675c0ed52ef1809425f0c77d9d2fd39b9e8/CMakeLists.txt#L11-L19) and it works fine to me.
If I do not set `TBB_TEST` to OFF, I have some compilation issues on tbb tests:
```bash
[  1%] Building CXX object _deps/tbb-build/test/CMakeFiles/_test_malloc_lib_unload.dir/tbbmalloc/test_malloc_lib_unload.cpp.o
[  1%] Building CXX object _deps/tbb-build/src/tbbmalloc/CMakeFiles/tbbmalloc.dir/backend.cpp.o
[  1%] Building CXX object _deps/tbb-build/test/CMakeFiles/test_handle_perror.dir/tbb/test_handle_perror.cpp.o
[  1%] Building CXX object _deps/tbb-build/src/tbbmalloc/CMakeFiles/tbbmalloc.dir/frontend.cpp.o
[  1%] Building CXX object _deps/tbb-build/src/tbbmalloc/CMakeFiles/tbbmalloc.dir/large_objects.cpp.o
[  2%] Building CXX object _deps/tbb-build/src/tbbmalloc/CMakeFiles/tbbmalloc.dir/__/tbb/itt_notify.cpp.o
[  3%] Building CXX object _deps/tbb-build/src/tbbmalloc/CMakeFiles/tbbmalloc.dir/tbbmalloc.cpp.o
[  3%] Building CXX object _deps/tbb-build/src/tbb/CMakeFiles/tbb.dir/arena.cpp.o
[  3%] Building CXX object _deps/tbb-build/test/CMakeFiles/test_malloc_whitebox.dir/tbbmalloc/test_malloc_whitebox.cpp.o
[  4%] Building CXX object _deps/tbb-build/src/tbb/CMakeFiles/tbb.dir/arena_slot.cpp.o
[  4%] Building CXX object _deps/tbb-build/src/tbb/CMakeFiles/tbb.dir/address_waiter.cpp.o
[  4%] Building CXX object _deps/tbb-build/src/tbb/CMakeFiles/tbb.dir/concurrent_bounded_queue.cpp.o
[  4%] Building CXX object _deps/tbb-build/src/tbbmalloc/CMakeFiles/tbbmalloc.dir/backref.cpp.o
[  5%] Building CXX object _deps/tbb-build/src/tbb/CMakeFiles/tbb.dir/dynamic_link.cpp.o
[  5%] Building CXX object _deps/tbb-build/src/tbb/CMakeFiles/tbb.dir/allocator.cpp.o
[  5%] Building CXX object _deps/tbb-build/src/tbb/CMakeFiles/tbb.dir/exception.cpp.o
[  5%] Building CXX object _deps/tbb-build/src/tbb/CMakeFiles/tbb.dir/governor.cpp.o
[  6%] Linking CXX shared library ../../../gnu_12.2_cxx17_64_relwithdebinfo/lib_test_malloc_lib_unload.so
[  6%] Built target _test_malloc_lib_unload
[  6%] Building CXX object _deps/tbb-build/test/CMakeFiles/test_malloc_lib_unload.dir/tbbmalloc/test_malloc_lib_unload.cpp.o
[  7%] Building CXX object _deps/tbb-build/src/tbb/CMakeFiles/tbb.dir/global_control.cpp.o
[  7%] Building CXX object _deps/tbb-build/src/tbb/CMakeFiles/tbb.dir/itt_notify.cpp.o
[  7%] Building CXX object _deps/tbb-build/src/tbb/CMakeFiles/tbb.dir/main.cpp.o
[  8%] Building CXX object _deps/tbb-build/src/tbb/CMakeFiles/tbb.dir/market.cpp.o
[  8%] Building CXX object _deps/tbb-build/src/tbb/CMakeFiles/tbb.dir/misc.cpp.o
[  8%] Building CXX object _deps/tbb-build/src/tbb/CMakeFiles/tbb.dir/misc_ex.cpp.o
[  9%] Building CXX object _deps/tbb-build/src/tbb/CMakeFiles/tbb.dir/observer_proxy.cpp.o
[  9%] Building CXX object _deps/tbb-build/src/tbb/CMakeFiles/tbb.dir/parallel_pipeline.cpp.o
[  9%] Building CXX object _deps/tbb-build/src/tbb/CMakeFiles/tbb.dir/private_server.cpp.o
[ 10%] Building CXX object _deps/tbb-build/src/tbb/CMakeFiles/tbb.dir/profiling.cpp.o
[ 10%] Linking C shared library ../../../../gnu_12.2_cxx17_64_relwithdebinfo/libtbbmalloc.so
[ 10%] Building CXX object _deps/tbb-build/src/tbb/CMakeFiles/tbb.dir/rml_tbb.cpp.o
[ 10%] Building CXX object _deps/tbb-build/src/tbb/CMakeFiles/tbb.dir/rtm_mutex.cpp.o
[ 10%] Building CXX object _deps/tbb-build/src/tbb/CMakeFiles/tbb.dir/rtm_rw_mutex.cpp.o
[ 11%] Building CXX object _deps/tbb-build/src/tbb/CMakeFiles/tbb.dir/semaphore.cpp.o
[ 11%] Building CXX object _deps/tbb-build/src/tbb/CMakeFiles/tbb.dir/small_object_pool.cpp.o
[ 11%] Building CXX object _deps/tbb-build/src/tbb/CMakeFiles/tbb.dir/task.cpp.o
[ 12%] Building CXX object _deps/tbb-build/src/tbb/CMakeFiles/tbb.dir/task_dispatcher.cpp.o
[ 12%] Building CXX object _deps/tbb-build/src/tbb/CMakeFiles/tbb.dir/task_group_context.cpp.o
[ 12%] Building CXX object _deps/tbb-build/src/tbb/CMakeFiles/tbb.dir/version.cpp.o
[ 13%] Building CXX object _deps/tbb-build/src/tbb/CMakeFiles/tbb.dir/queuing_rw_mutex.cpp.o
[ 13%] Built target tbbmalloc
[ 13%] Building CXX object _deps/tbb-build/test/CMakeFiles/test_malloc_regression.dir/tbbmalloc/test_malloc_regression.cpp.o
[ 14%] Building CXX object _deps/tbb-build/test/CMakeFiles/test_scalable_allocator.dir/tbbmalloc/test_scalable_allocator.cpp.o
[ 14%] Building CXX object _deps/tbb-build/src/tbbmalloc_proxy/CMakeFiles/tbbmalloc_proxy.dir/function_replacement.cpp.o
[ 15%] Building CXX object _deps/tbb-build/test/CMakeFiles/test_malloc_pools.dir/tbbmalloc/test_malloc_pools.cpp.o
[ 15%] Building CXX object _deps/tbb-build/test/CMakeFiles/test_malloc_compliance.dir/tbbmalloc/test_malloc_compliance.cpp.o
[ 15%] Building CXX object _deps/tbb-build/test/CMakeFiles/test_malloc_init_shutdown.dir/tbbmalloc/test_malloc_init_shutdown.cpp.o
[ 16%] Building CXX object _deps/tbb-build/test/CMakeFiles/_test_malloc_used_by_lib.dir/tbbmalloc/test_malloc_used_by_lib.cpp.o
[ 16%] Building C object _deps/tbb-build/test/CMakeFiles/test_malloc_pure_c.dir/tbbmalloc/test_malloc_pure_c.c.o
[ 16%] Building CXX object _deps/tbb-build/src/tbbmalloc_proxy/CMakeFiles/tbbmalloc_proxy.dir/proxy.cpp.o
[ 17%] Linking C executable ../../../gnu_12.2_cxx17_64_relwithdebinfo/test_malloc_pure_c
[ 17%] Built target test_malloc_pure_c
[ 17%] Linking CXX shared library ../../../../gnu_12.2_cxx17_64_relwithdebinfo/libtbb.so
[ 17%] Linking CXX shared library ../../../gnu_12.2_cxx17_64_relwithdebinfo/lib_test_malloc_used_by_lib.so
[ 17%] Built target _test_malloc_used_by_lib
[ 17%] Building CXX object _deps/tbb-build/test/CMakeFiles/test_malloc_used_by_lib.dir/tbbmalloc/test_malloc_used_by_lib.cpp.o
[ 17%] Linking CXX shared library ../../../../gnu_12.2_cxx17_64_relwithdebinfo/libtbbmalloc_proxy.so
[ 17%] Built target tbbmalloc_proxy
[ 17%] Building CXX object _deps/tbb-build/test/CMakeFiles/_test_malloc_atexit.dir/tbbmalloc/test_malloc_atexit.cpp.o
[ 18%] Building CXX object _deps/tbb-build/test/CMakeFiles/test_malloc_new_handler.dir/tbbmalloc/test_malloc_new_handler.cpp.o
[ 18%] Building CXX object _deps/tbb-build/test/CMakeFiles/test_malloc_overload_disable.dir/tbbmalloc/test_malloc_overload_disable.cpp.o
[ 18%] Building CXX object _deps/tbb-build/test/CMakeFiles/test_malloc_overload.dir/tbbmalloc/test_malloc_overload.cpp.o
[ 18%] Linking CXX shared library ../../../gnu_12.2_cxx17_64_relwithdebinfo/lib_test_malloc_atexit.so
[ 18%] Built target _test_malloc_atexit
[ 18%] Building CXX object _deps/tbb-build/test/CMakeFiles/test_malloc_atexit.dir/tbbmalloc/test_malloc_atexit.cpp.o
[ 18%] Built target tbb
[ 19%] Building CXX object _deps/tbb-build/test/CMakeFiles/test_tick_count.dir/tbb/test_tick_count.cpp.o
[ 20%] Building CXX object CMakeFiles/one_tbb_example.dir/one_tbb_example.cpp.o
[ 20%] Building CXX object _deps/tbb-build/test/CMakeFiles/test_allocators.dir/tbb/test_allocators.cpp.o
[ 21%] Linking CXX executable ../../../gnu_12.2_cxx17_64_relwithdebinfo/test_handle_perror
[ 21%] Built target test_handle_perror
[ 22%] Building CXX object _deps/tbb-build/test/CMakeFiles/test_arena_priorities.dir/tbb/test_arena_priorities.cpp.o
[ 22%] Linking CXX executable one_tbb_example
[ 22%] Built target one_tbb_example
[ 23%] Building CXX object _deps/tbb-build/test/CMakeFiles/test_dynamic_link.dir/tbb/test_dynamic_link.cpp.o
[ 23%] Linking CXX executable ../../../gnu_12.2_cxx17_64_relwithdebinfo/test_malloc_lib_unload
[ 23%] Built target test_malloc_lib_unload
[ 24%] Building CXX object _deps/tbb-build/test/CMakeFiles/test_collaborative_call_once.dir/tbb/test_collaborative_call_once.cpp.o
In file included from onetbb_cmake_fetch_content/build/_deps/tbb-src/src/tbb/../../include/oneapi/tbb/concurrent_vector.h:24,
                 from onetbb_cmake_fetch_content/build/_deps/tbb-src/src/tbb/../../include/oneapi/tbb/enumerable_thread_specific.h:26,
                 from onetbb_cmake_fetch_content/build/_deps/tbb-src/test/common/utils_concurrency_limit.h:25,
                 from onetbb_cmake_fetch_content/build/_deps/tbb-src/test/tbb/test_tick_count.cpp:19:
onetbb_cmake_fetch_content/build/_deps/tbb-src/src/tbb/../../include/oneapi/tbb/detail/_segment_table.h: In instantiation of 'tbb::detail::d1::segment_table<T, Allocator, DerivedType, PointersPerEmbeddedTable>::segment_table(const allocator_type&) [with T = tbb::detail::d0::padded<tbb::detail::d1::ets_element<bool>, 128>; Allocator = tbb::detail::d1::cache_aligned_allocator<tbb::detail::d0::padded<tbb::detail::d1::ets_element<bool>, 128> >; DerivedType = tbb::detail::d1::concurrent_vector<tbb::detail::d0::padded<tbb::detail::d1::ets_element<bool>, 128>, tbb::detail::d1::cache_aligned_allocator<tbb::detail::d0::padded<tbb::detail::d1::ets_element<bool>, 128> > >; long unsigned int PointersPerEmbeddedTable = 3; allocator_type = tbb::detail::d1::cache_aligned_allocator<tbb::detail::d0::padded<tbb::detail::d1::ets_element<bool>, 128> >]':
onetbb_cmake_fetch_content/build/_deps/tbb-src/src/tbb/../../include/oneapi/tbb/concurrent_vector.h:284:26:   required from 'tbb::detail::d1::concurrent_vector<T, Allocator>::concurrent_vector(const allocator_type&) [with T = tbb::detail::d0::padded<tbb::detail::d1::ets_element<bool>, 128>; Allocator = tbb::detail::d1::cache_aligned_allocator<tbb::detail::d0::padded<tbb::detail::d1::ets_element<bool>, 128> >; allocator_type = tbb::detail::d1::cache_aligned_allocator<tbb::detail::d0::padded<tbb::detail::d1::ets_element<bool>, 128> >]'
onetbb_cmake_fetch_content/build/_deps/tbb-src/src/tbb/../../include/oneapi/tbb/concurrent_vector.h:281:61:   required from 'tbb::detail::d1::concurrent_vector<T, Allocator>::concurrent_vector() [with T = tbb::detail::d0::padded<tbb::detail::d1::ets_element<bool>, 128>; Allocator = tbb::detail::d1::cache_aligned_allocator<tbb::detail::d0::padded<tbb::detail::d1::ets_element<bool>, 128> >]'
onetbb_cmake_fetch_content/build/_deps/tbb-src/src/tbb/../../include/oneapi/tbb/enumerable_thread_specific.h:843:5:   required from 'tbb::detail::d1::enumerable_thread_specific<T, Allocator, ETS_key_type>::enumerable_thread_specific() [with T = bool; Allocator = tbb::detail::d1::cache_aligned_allocator<bool>; tbb::detail::d1::ets_key_usage_type ETS_key_type = tbb::detail::d1::ets_no_key]'
onetbb_cmake_fetch_content/build/_deps/tbb-src/test/common/utils_concurrency_limit.h:247:92:   required from here
onetbb_cmake_fetch_content/build/_deps/tbb-src/src/tbb/../../include/oneapi/tbb/detail/_segment_table.h:63:63: error: member 'tbb::detail::d1::segment_table<tbb::detail::d0::padded<tbb::detail::d1::ets_element<bool>, 128>, tbb::detail::d1::cache_aligned_allocator<tbb::detail::d0::padded<tbb::detail::d1::ets_element<bool>, 128> >, tbb::detail::d1::concurrent_vector<tbb::detail::d0::padded<tbb::detail::d1::ets_element<bool>, 128>, tbb::detail::d1::cache_aligned_allocator<tbb::detail::d0::padded<tbb::detail::d1::ets_element<bool>, 128> > >, 3>::my_embedded_table' is used uninitialized [-Werror=uninitialized]
   63 |         : my_segment_table_allocator(alloc), my_segment_table(my_embedded_table)
      |                                                               ^~~~~~~~~~~~~~~~~
compilation terminated due to -Wfatal-errors.
cc1plus: all warnings being treated as errors
make[2]: *** [_deps/tbb-build/test/CMakeFiles/test_tick_count.dir/build.make:76: _deps/tbb-build/test/CMakeFiles/test_tick_count.dir/tbb/test_tick_count.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:613: _deps/tbb-build/test/CMakeFiles/test_tick_count.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
In file included from onetbb_cmake_fetch_content/build/_deps/tbb-src/src/tbb/../../include/oneapi/tbb/concurrent_vector.h:24,
                 from onetbb_cmake_fetch_content/build/_deps/tbb-src/src/tbb/../../include/oneapi/tbb/enumerable_thread_specific.h:26,
                 from onetbb_cmake_fetch_content/build/_deps/tbb-src/test/common/utils_concurrency_limit.h:25,
                 from onetbb_cmake_fetch_content/build/_deps/tbb-src/test/tbb/test_arena_priorities.cpp:25:
onetbb_cmake_fetch_content/build/_deps/tbb-src/src/tbb/../../include/oneapi/tbb/detail/_segment_table.h: In instantiation of 'tbb::detail::d1::segment_table<T, Allocator, DerivedType, PointersPerEmbeddedTable>::segment_table(const allocator_type&) [with T = tbb::detail::d0::padded<tbb::detail::d1::ets_element<bool>, 128>; Allocator = tbb::detail::d1::cache_aligned_allocator<tbb::detail::d0::padded<tbb::detail::d1::ets_element<bool>, 128> >; DerivedType = tbb::detail::d1::concurrent_vector<tbb::detail::d0::padded<tbb::detail::d1::ets_element<bool>, 128>, tbb::detail::d1::cache_aligned_allocator<tbb::detail::d0::padded<tbb::detail::d1::ets_element<bool>, 128> > >; long unsigned int PointersPerEmbeddedTable = 3; allocator_type = tbb::detail::d1::cache_aligned_allocator<tbb::detail::d0::padded<tbb::detail::d1::ets_element<bool>, 128> >]':
onetbb_cmake_fetch_content/build/_deps/tbb-src/src/tbb/../../include/oneapi/tbb/concurrent_vector.h:284:26:   required from 'tbb::detail::d1::concurrent_vector<T, Allocator>::concurrent_vector(const allocator_type&) [with T = tbb::detail::d0::padded<tbb::detail::d1::ets_element<bool>, 128>; Allocator = tbb::detail::d1::cache_aligned_allocator<tbb::detail::d0::padded<tbb::detail::d1::ets_element<bool>, 128> >; allocator_type = tbb::detail::d1::cache_aligned_allocator<tbb::detail::d0::padded<tbb::detail::d1::ets_element<bool>, 128> >]'
onetbb_cmake_fetch_content/build/_deps/tbb-src/src/tbb/../../include/oneapi/tbb/concurrent_vector.h:281:61:   required from 'tbb::detail::d1::concurrent_vector<T, Allocator>::concurrent_vector() [with T = tbb::detail::d0::padded<tbb::detail::d1::ets_element<bool>, 128>; Allocator = tbb::detail::d1::cache_aligned_allocator<tbb::detail::d0::padded<tbb::detail::d1::ets_element<bool>, 128> >]'
onetbb_cmake_fetch_content/build/_deps/tbb-src/src/tbb/../../include/oneapi/tbb/enumerable_thread_specific.h:843:5:   required from 'tbb::detail::d1::enumerable_thread_specific<T, Allocator, ETS_key_type>::enumerable_thread_specific() [with T = bool; Allocator = tbb::detail::d1::cache_aligned_allocator<bool>; tbb::detail::d1::ets_key_usage_type ETS_key_type = tbb::detail::d1::ets_no_key]'
onetbb_cmake_fetch_content/build/_deps/tbb-src/test/common/utils_concurrency_limit.h:247:92:   required from here
onetbb_cmake_fetch_content/build/_deps/tbb-src/src/tbb/../../include/oneapi/tbb/detail/_segment_table.h:63:63: error: member 'tbb::detail::d1::segment_table<tbb::detail::d0::padded<tbb::detail::d1::ets_element<bool>, 128>, tbb::detail::d1::cache_aligned_allocator<tbb::detail::d0::padded<tbb::detail::d1::ets_element<bool>, 128> >, tbb::detail::d1::concurrent_vector<tbb::detail::d0::padded<tbb::detail::d1::ets_element<bool>, 128>, tbb::detail::d1::cache_aligned_allocator<tbb::detail::d0::padded<tbb::detail::d1::ets_element<bool>, 128> > >, 3>::my_embedded_table' is used uninitialized [-Werror=uninitialized]
   63 |         : my_segment_table_allocator(alloc), my_segment_table(my_embedded_table)
      |                                                               ^~~~~~~~~~~~~~~~~
compilation terminated due to -Wfatal-errors.
cc1plus: all warnings being treated as errors
make[2]: *** [_deps/tbb-build/test/CMakeFiles/test_arena_priorities.dir/build.make:76: _deps/tbb-build/test/CMakeFiles/test_arena_priorities.dir/tbb/test_arena_priorities.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:665: _deps/tbb-build/test/CMakeFiles/test_arena_priorities.dir/all] Error 2
[ 24%] Linking CXX executable ../../../gnu_12.2_cxx17_64_relwithdebinfo/test_malloc_overload_disable
[ 24%] Built target test_malloc_overload_disable
In file included from onetbb_cmake_fetch_content/build/_deps/tbb-src/src/tbb/../../include/oneapi/tbb/concurrent_vector.h:24,
                 from onetbb_cmake_fetch_content/build/_deps/tbb-src/src/tbb/../../include/oneapi/tbb/enumerable_thread_specific.h:26,
                 from onetbb_cmake_fetch_content/build/_deps/tbb-src/test/common/utils_concurrency_limit.h:25,
                 from onetbb_cmake_fetch_content/build/_deps/tbb-src/test/tbb/test_collaborative_call_once.cpp:34:
onetbb_cmake_fetch_content/build/_deps/tbb-src/src/tbb/../../include/oneapi/tbb/detail/_segment_table.h: In instantiation of 'tbb::detail::d1::segment_table<T, Allocator, DerivedType, PointersPerEmbeddedTable>::segment_table(const allocator_type&) [with T = tbb::detail::d0::padded<tbb::detail::d1::ets_element<bool>, 128>; Allocator = tbb::detail::d1::cache_aligned_allocator<tbb::detail::d0::padded<tbb::detail::d1::ets_element<bool>, 128> >; DerivedType = tbb::detail::d1::concurrent_vector<tbb::detail::d0::padded<tbb::detail::d1::ets_element<bool>, 128>, tbb::detail::d1::cache_aligned_allocator<tbb::detail::d0::padded<tbb::detail::d1::ets_element<bool>, 128> > >; long unsigned int PointersPerEmbeddedTable = 3; allocator_type = tbb::detail::d1::cache_aligned_allocator<tbb::detail::d0::padded<tbb::detail::d1::ets_element<bool>, 128> >]':
onetbb_cmake_fetch_content/build/_deps/tbb-src/src/tbb/../../include/oneapi/tbb/concurrent_vector.h:284:26:   required from 'tbb::detail::d1::concurrent_vector<T, Allocator>::concurrent_vector(const allocator_type&) [with T = tbb::detail::d0::padded<tbb::detail::d1::ets_element<bool>, 128>; Allocator = tbb::detail::d1::cache_aligned_allocator<tbb::detail::d0::padded<tbb::detail::d1::ets_element<bool>, 128> >; allocator_type = tbb::detail::d1::cache_aligned_allocator<tbb::detail::d0::padded<tbb::detail::d1::ets_element<bool>, 128> >]'
onetbb_cmake_fetch_content/build/_deps/tbb-src/src/tbb/../../include/oneapi/tbb/concurrent_vector.h:281:61:   required from 'tbb::detail::d1::concurrent_vector<T, Allocator>::concurrent_vector() [with T = tbb::detail::d0::padded<tbb::detail::d1::ets_element<bool>, 128>; Allocator = tbb::detail::d1::cache_aligned_allocator<tbb::detail::d0::padded<tbb::detail::d1::ets_element<bool>, 128> >]'
onetbb_cmake_fetch_content/build/_deps/tbb-src/src/tbb/../../include/oneapi/tbb/enumerable_thread_specific.h:843:5:   required from 'tbb::detail::d1::enumerable_thread_specific<T, Allocator, ETS_key_type>::enumerable_thread_specific() [with T = bool; Allocator = tbb::detail::d1::cache_aligned_allocator<bool>; tbb::detail::d1::ets_key_usage_type ETS_key_type = tbb::detail::d1::ets_no_key]'
onetbb_cmake_fetch_content/build/_deps/tbb-src/test/common/utils_concurrency_limit.h:247:92:   required from here
onetbb_cmake_fetch_content/build/_deps/tbb-src/src/tbb/../../include/oneapi/tbb/detail/_segment_table.h:63:63: error: member 'tbb::detail::d1::segment_table<tbb::detail::d0::padded<tbb::detail::d1::ets_element<bool>, 128>, tbb::detail::d1::cache_aligned_allocator<tbb::detail::d0::padded<tbb::detail::d1::ets_element<bool>, 128> >, tbb::detail::d1::concurrent_vector<tbb::detail::d0::padded<tbb::detail::d1::ets_element<bool>, 128>, tbb::detail::d1::cache_aligned_allocator<tbb::detail::d0::padded<tbb::detail::d1::ets_element<bool>, 128> > >, 3>::my_embedded_table' is used uninitialized [-Werror=uninitialized]
   63 |         : my_segment_table_allocator(alloc), my_segment_table(my_embedded_table)
      |                                                               ^~~~~~~~~~~~~~~~~
compilation terminated due to -Wfatal-errors.
cc1plus: all warnings being treated as errors
make[2]: *** [_deps/tbb-build/test/CMakeFiles/test_collaborative_call_once.dir/build.make:76: _deps/tbb-build/test/CMakeFiles/test_collaborative_call_once.dir/tbb/test_collaborative_call_once.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:717: _deps/tbb-build/test/CMakeFiles/test_collaborative_call_once.dir/all] Error 2
[ 24%] Linking CXX executable ../../../gnu_12.2_cxx17_64_relwithdebinfo/test_malloc_init_shutdown
[ 24%] Built target test_malloc_init_shutdown
[ 24%] Linking CXX executable ../../../gnu_12.2_cxx17_64_relwithdebinfo/test_malloc_used_by_lib
[ 24%] Built target test_malloc_used_by_lib
[ 25%] Linking CXX executable ../../../gnu_12.2_cxx17_64_relwithdebinfo/test_malloc_atexit
[ 25%] Built target test_malloc_atexit
[ 25%] Linking CXX executable ../../../gnu_12.2_cxx17_64_relwithdebinfo/test_malloc_regression
[ 25%] Built target test_malloc_regression
[ 25%] Linking CXX executable ../../../gnu_12.2_cxx17_64_relwithdebinfo/test_malloc_new_handler
[ 25%] Built target test_malloc_new_handler
[ 26%] Linking CXX executable ../../../gnu_12.2_cxx17_64_relwithdebinfo/test_malloc_overload
[ 26%] Built target test_malloc_overload
[ 26%] Linking CXX executable ../../../gnu_12.2_cxx17_64_relwithdebinfo/test_malloc_compliance
[ 26%] Built target test_malloc_compliance
[ 26%] Linking CXX executable ../../../gnu_12.2_cxx17_64_relwithdebinfo/test_malloc_whitebox
[ 26%] Linking CXX executable ../../../gnu_12.2_cxx17_64_relwithdebinfo/test_malloc_pools
[ 26%] Built target test_malloc_whitebox
[ 26%] Built target test_malloc_pools
[ 26%] Linking CXX executable ../../../gnu_12.2_cxx17_64_relwithdebinfo/test_dynamic_link
[ 26%] Built target test_dynamic_link
[ 27%] Linking CXX executable ../../../gnu_12.2_cxx17_64_relwithdebinfo/test_allocators
[ 27%] Built target test_allocators
[ 28%] Linking CXX executable ../../../gnu_12.2_cxx17_64_relwithdebinfo/test_scalable_allocator
[ 28%] Built target test_scalable_allocator
make: *** [Makefile:166: all] Error 2
```

**Comment by isaevil** [comment]
@VincentRouvreau I see you are using v2021.7. Could you please check also with v2021.8?

**Comment by VincentRouvreau** [comment]
Same errors with v2021.8.

**Comment by isaevil** [comment]
@VincentRouvreau these errors are fixed by #958. You can expect this fixed in one of future releases or you can just use recent master branch.

**Comment by isaevil** [comment]
Looks like initial issue can be solved by using `option(...)` as proposed.
@VincentRouvreau @HEUBITLYJ closing this as resolved. Feel free to reopen if issue still exists.
